pilot/pkg/bootstrap/server.go

```diff
@@ -532,6 +532,7 @@ func (s *Server) initMCPConfigController(args *PilotArgs) error {
 						log.Infof("%v not found. Checking again in %v", requiredFiles[0], requiredMCPCertCheckFreq)
 						select {
 						case <-ctx.Done():
+							cancel()
```

Comment:
This context is shared across multiple configuration sources. Should one failed client connection cause _all_ connections to fail? Or should we fail more gracefully and keep the other connections alive?

Reply:
Actually this does not matter too much, because one err will result in process exit.